internal/run/source.go

```diff
@@ -194,11 +194,13 @@ func (w *Workflow) RunSource(ctx context.Context, parentStep *workflowTracking.W
 	// If the source has a previous tracked revision, compute changes against it
 	if w.lockfileOld != nil && !w.SkipChangeReport {
 		if targetLockOld, ok := w.lockfileOld.Targets[targetID]; ok && !utils.IsZeroTelemetryOrganization(ctx) {
-			sourceRes.ChangeReport, err = w.computeChanges(ctx, rootStep, targetLockOld, currentDocument)
+			report, specInfo, err := w.computeChanges(ctx, rootStep, targetLockOld, currentDocument)
 			if err != nil {
 				// Don't fail the whole workflow if this fails
 				logger.Warnf("failed to compute OpenAPI changes: %s", err.Error())
 			}
+			sourceRes.ChangeReport = report
+			sourceMap[sourceID] = specInfo
```

Comment:
why not store this in the `sourceRes`? The pattern of passing in a variable that you mutate should be used very sparingly in my opinion